daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/TypeCalculator.scala

```diff
@@ -176,17 +176,17 @@ class KeysetValueTypeCalculatorUnordered(valueMap: HashMap[DataValuePrimitive, D
     if (valueMap.contains(x)) {
       valueMap.get(x) match {
         case Some(a) => (a, Maybe.Nope)
-        case None => (DataValue.NoValue, One(s"Value ${x} not found in keyset-value mapping"))
+        case None => (DataValue.NoValue, One(s"Value ${x} not found in enumeration key-value mapping"))
       }
     } else {
-      (DataValue.NoValue, One(s"Key ${x} not found in keyset-value mapping"))
+      (DataValue.NoValue, One(s"Key ${x} not found in enumeration key-value mapping"))
 
     }
   }
   override def outputTypeCalc(x: DataValuePrimitive, xType: NodeInfo.Kind): (DataValuePrimitiveNullable, Maybe[Error]) = {
     unparseMap.get(x) match {
       case Some(v) => (v, Maybe.Nope)
-      case None => (DataValue.NoValue, One(s"Value ${x} not found in keyset-value mapping"))
+      case None => (DataValue.NoValue, One(s"Value ${x} not found in enumeration key-value mapping"))
```

Comment:
I wonder if we can be a bit more precise and thus more helpful? My concern is "key-value mapping" isn't a term a schema developer might know since none of the properties use those terms.

I think when parsing we are looking for a value in `dfdlx:repValues`, and when unparsing we are looking for a string in the enumeration `value`? Is it worth differentiating and saying something like this when parsing:

> Value ${x} not found in enumeration dfdlx:repValues properties

And this for unparsing:

> Value ${x} not found in enumeration value properties

I'm not sure I like that wording a ton better, but I think mentioning value/repValues might help to point the user at the right attribute that we are looking for a match but didn't find?

Reply:
You can stick "(keyset-value mapping)" at the end of the messages to maintain backward compatibility with past message-based tests maybe. So long as this isn't needed to understand the primary diagnostic message. I agree with SL that this should mention enumeration, mention dfdlx:repValues (I would not search for the dfdlx prefix in tests though since that's likely to go away some day), and mention being unable to determine the repValue for the unrecognized enumeration value for unparsing.

Reply:
As far as I can tell, the only place referencing the keyset-mapping error message is like 2 tests in P8, which I was about to update anyway with some small changes from SBSA, so I'll just update the tests there.  None of the other schemas in the regression test repo reference that error message.